projects/cm_mcu/AlarmUtilities.c

```diff
@@ -201,84 +203,88 @@ int VoltStatus(void)
   }
 
 #ifdef REV2
-  while (getADCvalue(ADC_INFO_GEN_VCC_4V0_CH) < 0.01f) { // somehow the initial value of VCC 4V0 is 0 after reboot and can screw up the logic below
-    vTaskDelay(pdMS_TO_TICKS(10));                       // delay 10 ms
-  }
+  // while (getADCvalue(ADC_INFO_GEN_VCC_4V0_CH) < 0.5f) { // somehow the initial value of VCC 4V0 is 0 after reboot and can screw up the logic below
+  // vTaskDelay(pdMS_TO_TICKS(10));                       // delay 10 ms
+  //}
 #endif
 
   for (int ch = ADC_INFO_GEN_VCC_INIT_CH; ch < ADC_INFO_GEN_VCC_FIN_CH + 1; ++ch) {
 
+    while (getADCvalue(ch) < 0.5f) {  // wait for delay from ADC outputs and actual reading
+      vTaskDelay(pdMS_TO_TICKS(100)); // delay 100 ms
+    }
     float threshold = getAlarmVoltageThres();
     float now_value = getADCvalue(ch);
     float excess = (now_value - getADCtargetValue(ch)) / getADCtargetValue(ch);
     int tens, frac;
     float_to_ints(excess * 100, &tens, &frac);
     if (excess > 0.0f) {
-      is_alarm_volt = 1;
+      is_gen_alarm_volt = 1;
       excess_volt = excess * 100;
       excess_volt_which_ch = ch;
     }
 
     if ((excess > threshold && excess > 0.0f) || (excess * -1.0f > threshold && excess < 0.0f)) { // if this ADC voltage is greater/lower than a target value by getAlarmVoltageThres()*100%
       gen_bitmask += (1 << (ch - ADC_INFO_GEN_VCC_INIT_CH));                                      // first to last bit corresponds to status of low to high ADC voltage channel of mcu/general
-      is_alarm_volt = 2;
+      is_gen_alarm_volt = 2;
       log_debug(LOG_ALM, "alarm volt at ADC ch : %02d now %02d.%02d %% off target\r\n", ch, tens, frac); // over voltage among one of fpga power supplies by +/- getAlarmVoltageThres()*100% of its threshold
     }
   }
 
-  if (is_alarm_volt > 0) {
-    retval++;
-    if (is_alarm_volt == 2) {
-      status_V |= ALM_STAT_GEN_OVERVOLT;
-      ++retval;
-    }
-  }
-
   currentVoltStatus[GEN] = gen_bitmask & GEN_VOLTAGE_MASK; // applies a mask with power-off exceptions
 
-  if (retval > 0)
-    return retval;
-  else
-    retval = 0;
-
-  is_alarm_volt = 0;
-  status_V = 0x0U;
-
   int n_fpga_half_ch = (ADC_INFO_FPGA_VCC_FIN_CH - ADC_INFO_FPGA_VCC_INIT_CH + 1) / 2;
   int ADC_INFO_FPGA2_VCC_INIT_CH = n_fpga_half_ch + ADC_INFO_FPGA_VCC_INIT_CH;
   for (int ch = ADC_INFO_FPGA_VCC_INIT_CH; ch < ADC_INFO_FPGA_VCC_FIN_CH + 1; ++ch) {
     if ((!f1_enable) || (!f2_enable && ch > (ADC_INFO_FPGA2_VCC_INIT_CH - 1))) // check if fpga1/2 is on the board. currently fpga1 takes the first half of adc outputs in this indexing
       break;
+    while (getADCvalue(ch) < 0.5f) {  // wait for delay from ADC outputs and actual reading
+      vTaskDelay(pdMS_TO_TICKS(100)); // delay 100 ms
+    }
```

Comment:
what is this supposed to do? Won't this cause an infinite loop when the power is off?

Reply:
it actually does harm more than good just as you reasoned. The idea was to wait for the ADC outputs to sync with the actual voltage values. I think I might have forgotten to remove it after I have added a condition to printout a warning for excessive voltage to be more than 2%. If you remember, there used to be an annoying voltage warning after boost with excess voltage < 1%. 

removed. Done.